tests/run.rs

```diff
@@ -45,6 +50,50 @@ fn simple() -> R<()> {
     Ok(())
 }
 
+mod yaml_parse_errors {
+    use super::*;
+
+    #[test]
+    fn wrong_types() -> R<()> {
+        let script = TempFile::write_temp_script("")?;
+        let result = test_run_check_protocols(
+            &script,
+            r##"
+                |protocol: 42
+            "##,
+        );
+        assert_eq!(
+            format!("{}", result.unwrap_err()),
```

Comment:
just curious, is this equivalent to `result.unwrap_err().to_string()`?

Reply:
Yes, it is. TIL. `format("{}", ...)` uses the `Display` trait, but `ToString` is automatically implemented for everything that implements `Display`. (https://doc.rust-lang.org/std/string/trait.ToString.html#tymethod.to_string)